frontend/src/app/components/selectors/SelectorBase.ts

```diff
@@ -70,22 +77,38 @@ export abstract class SelectorBase<CustomSliceSelection extends SliceSelection>
 
     public archiveSelections(selectionMap?: Array<CustomSliceSelection>): void {
         if (!selectionMap) {
-            selectionMap = Array.from(this.selections.values());
+            selectionMap = this.getSelections();
         }
         selectionMap.forEach((value: CustomSliceSelection) => {
             this.archivedSelections.push(value);
         });
     }
 
-    public removeCurrentSelection(): void {
-        if (this.hasSliceSelection()) {
+    public removeSelectionsOnCurrentSlice(): void {
+        if (this.selections.get(this.currentSlice)) {
```

Comment:
You can get selections one time to some variable, check if it exists and them use forEach.